go-controller/pkg/ovn/ovn.go

```diff
@@ -783,7 +783,7 @@ func (oc *Controller) syncNodeGateway(node *kapi.Node, hostSubnets []*net.IPNet)
 		var hostAddrs sets.String
 		if config.Gateway.Mode == config.GatewayModeShared {
 			hostAddrs, err = util.ParseNodeHostAddresses(node)
-			if err != nil {
+			if err != nil && !util.IsAnnotationNotSetError(err) {
```

Comment:
are you doing this because the smart nic stuff you added is not using node ip manager? If so, condition it here based on the config mode and dont change the normal behavior

Reply:
ok you convinced me :)